bin/build-plugin-zip.sh

```diff
@@ -25,34 +35,50 @@ elif ! git diff --cached --exit-code > /dev/null; then
 fi
 if [ ! -z "$changed" ]; then
 	git status
-	echo "ERROR: Cannot build plugin zip with dirty working tree."
-	echo "       Commit your changes and try again."
-	exit 1
+	error "ERROR: Cannot build plugin zip with dirty working tree. ☝️
+       Commit your changes and try again."
+	exit
 fi
 
 branch="$(git rev-parse --abbrev-ref HEAD)"
 if [ "$branch" != 'master' ]; then
-	echo "WARNING: You should probably be running this script against the"
-	echo "         'master' branch (current: '$branch')"
-	echo
+	warning "WARNING: You should probably be running this script against the
+         'master' branch (current: '$branch')"
 	sleep 2
 fi
 
-# Remove ignored files to reset repository to pristine condition. Previous test
-# ensures that changed files abort the plugin build.
-status "Cleaning working directory..."
-git clean -xdf
+# Do a dry run of the repository reset. Prompting the user for a list of all
+# files that will be removed should prevent them from losing important files!
+status "Resetting the repository to pristine condition."
+echo "Would remove gutenberg.zip"
+git clean -xdf --dry-run
+warning "🚨 About to delete everything above! Is this okay? 🚨"
+echo -n "[Y]es/[N]o: "
+read answer
+if [ "$answer" != "${answer#[Yy]}" ]; then
+	# Remove any existing zip file.
+	rm -f gutenberg.zip
```

Comment:
Wouldn't `git clean -xdf` take care of this file too?

Reply:
I wondered that and thought yes, but it was in there from before. It doesn't show up in the regular output, I'm guessing because it's in `.gitignore` so git "knows" about it?

Reply:
Scratch that, it *does* know about it. Dunno why the deletion was in there before; I'll remove this 🎉